cclient/iota_client_extended_api.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "iota_client_extended_api.h"
+
+static bool is_unused_address(iota_client_service_t const* const serv,
+                              flex_trit_t const* const addr) {
+  // retcode_t ret = RC_OK;
+  int ret_num = 0;
+  tryte_t trytes_addr[NUM_TRYTES_ADDRESS];
+  find_transactions_req_t* find_tran = find_transactions_req_new();
+  find_transactions_res_t* res = find_transactions_res_new();
+
+  // TODO remove flex_trits-trytes convertion.
+  flex_trits_to_trytes(trytes_addr, NUM_TRYTES_ADDRESS, addr, NUM_TRITS_ADDRESS,
+                       NUM_TRITS_ADDRESS);
+  find_tran = find_transactions_req_add_address(find_tran, (char*)trytes_addr);
+
+  iota_client_find_transactions(serv, find_tran, &res);
+
+  ret_num = find_transactions_res_hash_num(res);
+  return ret_num ? true : false;
```

Comment:
The function name is a bit misleading, it's called `is_unused_address ` but returns `true` if the address is used ?

Reply:
yeap..my bad.